src/main/java/org/reactome/release/qa/common/QACheckerHelper.java

```diff
@@ -225,5 +232,339 @@ public static String getAttributeTableName(String clsName,
+		Set<GKInstance> constituentPEs = new HashSet<>();
+		if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
+			constituentPEs.addAll(multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.hasComponent));
+		} else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
+			constituentPEs.addAll(multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.repeatedUnit));
+		}
+		return constituentPEs;
+	}
+
+	/**
+	 * Finds all PhysicalEntities that exist within an EntitySet by searching for all member PEs (via 'hasMember' attribute)
+	 * and all candidate PEs (if entitySet is a CandidateSet; via 'hasCandidate attribute).
+	 * @param entitySet GKInstance -- EntitySet instance that will be searched for all PhysicalEntities.
+	 * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in EntitySet.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	public static Set<GKInstance> findEntitySetPhysicalEntities(GKInstance entitySet) throws Exception {
+		Set<GKInstance> physicalEntities = new HashSet<>();
+		// All EntitySet instances have a 'hasMember' slot to be searched.
+		Set<String> entitySetAttributes = new HashSet<>(Arrays.asList(ReactomeJavaConstants.hasMember));
+		// If the EntitySet is a CandidateSet, 'hasCandidate' attribute needs to be searched as well.
+		if (entitySet.getSchemClass().isa(ReactomeJavaConstants.CandidateSet)) {
+			entitySetAttributes.add(ReactomeJavaConstants.hasCandidate);
+		}
+		for (String entitySetAttribute : entitySetAttributes) {
+			// Recursively searches members/candidates in EntitySet for all PhysicalEntities.
+			for (GKInstance setInstance : (Collection<GKInstance>) entitySet.getAttributeValuesList(entitySetAttribute)) {
+				physicalEntities.addAll(findAllPhysicalEntities((setInstance)));
+			}
+		}
+		return physicalEntities;
+	}
+
+	/**
+	 * Checks if incoming Event is manually inferred by checking inferredFrom referral.
+	 * @param event GKInstance -- Event instance being checked for inferredFrom referral.
+	 * @return boolean -- true if inferredFrom referral exists, false if not.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	public static boolean manuallyInferred(GKInstance event) throws Exception {
+		return event.getReferers(ReactomeJavaConstants.inferredFrom) != null;
+	}
+
+	/**
+	 * Checks if incoming DatabaseObject (Event or PhysicalEntity) has single Homo sapiens species.
+	 * @param databaseObject GKInstance -- Event or PhysicalEntity instance being checked for only Homo sapiens species.
+	 * @return boolean -- true if databaseObject only has a single, Homo sapiens species instance, false if not.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	public static boolean isHumanDatabaseObject(GKInstance databaseObject) throws Exception {
+		List<GKInstance> objectSpecies = databaseObject.getAttributeValuesList(ReactomeJavaConstants.species);
```

Comment:
Since this is a generic method, please check if the species is a valid attribute first.